src/common/arrow/arrow_appender.cpp

```diff
@@ -32,21 +34,25 @@ struct ArrowAppendData {
 	// child data (if any)
 	vector<unique_ptr<ArrowAppendData>> child_data;
 
-	//! the arrow array C API data, only set after Finalize
+	// the arrow array C API data, only set after Finalize
 	unique_ptr<ArrowArray> array;
 	duckdb::array<const void *, 3> buffers = {{nullptr, nullptr, nullptr}};
 	vector<ArrowArray *> child_pointers;
+
+	// If the data is exported using large buffers
+	bool export_large = true;
 };
 
 //===--------------------------------------------------------------------===//
 // ArrowAppender
 //===--------------------------------------------------------------------===//
-static unique_ptr<ArrowAppendData> InitializeArrowChild(const LogicalType &type, idx_t capacity);
+static unique_ptr<ArrowAppendData> InitializeArrowChild(const LogicalType &type, idx_t capacity, bool export_as_large);
```

Comment:
Could we also turn this into an enum (e.g. `ArrowOffsetSize::REGULAR`, `ArrowOffsetSize::LARGE`)